contextvh/src/main/java/contextvh/util/MapUtils.java

```diff
@@ -0,0 +1,159 @@
+package contextvh.util;
+
+import java.util.Arrays;
+import java.util.List;
+
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.MultiPolygon;
+import com.vividsolutions.jts.geom.prep.PreparedGeometry;
+import com.vividsolutions.jts.geom.prep.PreparedGeometryFactory;
+
+import nl.tytech.core.client.event.EventManager;
+import nl.tytech.core.net.serializable.MapLink;
+import nl.tytech.core.structure.ItemMap;
+import nl.tytech.data.engine.item.Building;
+import nl.tytech.data.engine.item.Land;
+import nl.tytech.data.engine.item.Setting;
+import nl.tytech.data.engine.item.Terrain;
+import nl.tytech.data.engine.item.Zone;
+import nl.tytech.data.engine.serializable.MapType;
+import nl.tytech.util.JTSUtils;
+
+/**
+ * Utility class for performing often used operations on map elements.
+ * @author Max Groenenboom
+ */
+public final class MapUtils {
+
+	/**
+	 * Private constructor.
+	 */
+	private MapUtils() {
+	}
+
+	protected static final MapType DEFAULT_MAPTYPE = MapType.MAQUETTE;
+
+	/**
+	 * Returns a Geometry of all zones with its ID in ids combined, if an empty list or no ids are given,
+	 * a Geometry containing all zones will be returned.
+	 * @param connectionID The id of the connection.
+	 * @param ids The list of zone ID's. If the list is empty, all zones will be used.
+	 * @return The resulting Geometry.
+	 */
+	public static Geometry getZonesCombined(final Integer connectionID, final List<Integer> ids) {
+		final boolean getAll = ids.size() == 0;
+		final ItemMap<Zone> zones = EventManager.getItemMap(connectionID, MapLink.ZONES);
+		Geometry result = JTSUtils.EMPTY;
+		for (Zone zone : zones) {
+			if (getAll || ids.contains(zone.getID())) {
+				result = result.union(zone.getMultiPolygon());
+			}
+		}
+		return result;
+	}
+
+	/**
+	 * Returns a Geometry of all zones with its ID in ids combined, if an empty list or no ids are given,
+	 * a Geometry containing all zones will be returned.
+	 * @param connectionID The id of the connection.
+	 * @param ids The list of zone ID's. If the list is empty, all zones will be used.
+	 * @return The resulting Geometry.
+	 */
+	public static Geometry getZonesCombined(final Integer connectionID, final Integer... ids) {
+		List<Integer> zoneIDs = Arrays.asList(ids);
+		return getZonesCombined(connectionID, zoneIDs);
+	}
+
+	/**
+	 * Returns a Geometry of all land of the stakeholder with ID stakeholderID.
+	 * @param connectionID The id of the connection.
+	 * @param stakeholderID The ID of the stakeholder.
+	 * @return The resulting Geometry.
+	 */
+	public static MultiPolygon getMyLands(final Integer connectionID, final Integer stakeholderID) {
```

Comment:
maybe call this getStakeholderLand as it is not actually specific to any stakeholder